### PR TITLE
docs: Update working-with-data-science-pipelines.adoc for improved readability

### PR DESCRIPTION
**Pull request by franciscojavierarceo**
## Description

Updating pipeline docs to enumerate pipeline components for readability

## How Has This Been Tested?
N/A

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by chtyler** [comment]
@franciscojavierarceo - Could you please explain a bit more why you think the content is less readable? The documentation team must ensure compliance with the IBM style guide, and also the Red Hat extended style guide. Because of this, I would like to comment on your proposed changes:

The steps that you have sequenced with a numbered list in your changes are not sequential. Therefore, it would make the content (arguably) less clear. This is the guidance that we have on sequential lists:

_Use an ordered (numbered) list when the sequence is significant, for example, when you write procedures or rank items_

Also we don't need to state the precise number of components, in this instance, in the introductory sentence to list.

File: working-with-data-science-pipelines.adoc [hunk 1]
```diff
@@ -38,17 +38,17 @@ Data science pipelines in {productname-short} are now based on link:https://www.
 endif::[]
 endif::[]
 
-A data science pipeline in {productname-short} consists of the following components:
+A data science pipeline in {productname-short} consists of the 4 following components:
 
-* Pipeline server: A server that is attached to your data science project and hosts your data science pipeline.
-* Pipeline: A pipeline defines the configuration of your machine learning workflow and the relationship between each component in the workflow.
+1. Pipeline server: A server that is attached to your data science project and hosts your data science pipeline.
```
Comment (franciscojavierarceo) [inline working-with-data-science-pipelines.adoc:43]:
```suggestion
* Pipeline server: A server that is attached to your data science project and hosts your data science pipeline.
```

File: working-with-data-science-pipelines.adoc [hunk 2]
```diff
@@ -38,17 +38,17 @@ Data science pipelines in {productname-short} are now based on link:https://www.
 endif::[]
 endif::[]
 
-A data science pipeline in {productname-short} consists of the following components:
+A data science pipeline in {productname-short} consists of the 4 following components:
 
 * Pipeline server: A server that is attached to your data science project and hosts your data science pipeline.
-* Pipeline: A pipeline defines the configuration of your machine learning workflow and the relationship between each component in the workflow.
+2. Pipeline: A pipeline defines the configuration of your machine learning workflow and the relationship between each component in the workflow.
```
Comment (franciscojavierarceo) [inline working-with-data-science-pipelines.adoc:44]:
```suggestion
* Pipeline: A pipeline defines the configuration of your machine learning workflow and the relationship between each component in the workflow.
```

File: working-with-data-science-pipelines.adoc [hunk 3]
```diff
@@ -38,17 +38,17 @@ Data science pipelines in {productname-short} are now based on link:https://www.
...
 
-A data science pipeline in {productname-short} consists of the following components:
+A data science pipeline in {productname-short} consists of the 4 following components:
 
 * Pipeline server: A server that is attached to your data science project and hosts your data science pipeline.
 * Pipeline: A pipeline defines the configuration of your machine learning workflow and the relationship between each component in the workflow.
 ** Pipeline code: A definition of your pipeline in a YAML file.
 ** Pipeline graph: A graphical illustration of the steps executed in a pipeline run and the relationship between them.
-* Pipeline experiment: A workspace where you can try different configurations of your pipelines. You can use experiments to organize your runs into logical groups.
+3. Pipeline experiment: A workspace where you can try different configurations of your pipelines. You can use experiments to organize your runs into logical groups.
```
Comment (franciscojavierarceo) [inline working-with-data-science-pipelines.adoc:47]:
```suggestion
* Pipeline experiment: A workspace where you can try different configurations of your pipelines. You can use experiments to organize your runs into logical groups.
```

File: working-with-data-science-pipelines.adoc [hunk 4]
```diff
@@ -48,7 +48,7 @@ A data science pipeline in {productname-short} consists of the following compone
 *** Archived pipeline experiment: An archived pipeline experiment.
 *** Pipeline artifact: An output artifact produced by a pipeline component. 
 *** Pipeline execution: The execution of a task in a pipeline.
-* Pipeline run: An execution of your pipeline.
+4. Pipeline run: An execution of your pipeline.
```
Comment (franciscojavierarceo) [inline working-with-data-science-pipelines.adoc:51]:
```suggestion
* Pipeline run: An execution of your pipeline.
```

**Comment by grainnejenningsRH** [comment]
Hi @franciscojavierarceo - adding the number of components here would create some inconsistency with how we word similar lines across the user guides (we don't tend to highlight the number of items in unordered lists). 

Alternatively, we can format the list items to make it clearer at a glance that there are 4 top-level items. e.g.:

Screenshot 2024-07-24 at 13 46 09

Would that help here?

**Comment by franciscojavierarceo** [comment]
Yeah I think so! This may be something that is particular to me so not overly opinionated on it!

**Comment by grainnejenningsRH** [comment]
Ok, good stuff. There are some related downstream areas in the docs that I'd like to make similar updates to, so I've created a ticket to cover all of these, and will close this PR until I can pull it into a sprint. Thanks for raising this! 👍

**Pull request closed without merging**